src/Common/memory.h

```diff
@@ -17,13 +17,28 @@
 #    include <cstdlib>
 #endif
 
+#if defined(OS_LINUX)
+#    include <malloc.h>
+#elif defined(OS_DARWIN)
+#    include <malloc/malloc.h>
+#endif
+
 namespace ProfileEvents
 {
     extern const Event GWPAsanAllocateSuccess;
     extern const Event GWPAsanAllocateFailed;
     extern const Event GWPAsanFree;
 }
 
+/// Guard pages interface.
+///
+/// Uses MADV_GUARD_INSTALL/MADV_GUARD_REMOVE which does not splits VMA, unlike mprotect()
+///
+/// Uses MADV_GUARD_INSTALL if available, or mprotect() if not
+void memoryGuardInstall(void *addr, size_t len);
```

Comment:
subjective thing, but IMO 

``` cpp
void installGuardPage(void * addr);
```

would be a little cleaner

Reply:
Yeah, it is always questionable...

I thought about your suggesting, but decided to keep as-is, since in the caller we explicitly use `getPageSize`, or even cache it, so it looks reasonable to use it

I also thought about introducing RAII, but does not make sense anyway, since all the places will need to try/catch then, does not looks better. That said that it is pretty low level interface and having RAII for it does not make sense.